content/blog/integrating-istio-and-spire.md

```diff
@@ -0,0 +1,463 @@
+
+Get the istio-spire-config patch using **[this link](https://raw.githubusercontent.com/cxteamtrials/caas-trials-content/main/services/istio/release-1.17/spire/spire-patch.yaml)** and copy that patch into your cluster. Install that patch using following command.
+
+```shellsession
+istioctl install -f istio-spire-config.yaml
+```
+
+This will share the spiffe-csi-driver with the Ingress Gateway and the sidecars that are going to be injected on workload pods, granting them access to the SPIRE Agent’s UNIX Domain Socket.
+
+* #### Patching Istio-Ingress gateways
+
+  If you receive the error shown below, your ingress-gateway is not patched yet and is not being registered onto the server.
+
+  ![](/img/patch-error-ingress.jpg)
+
+   For patching, the first step is to get and apply one of SPIRE controller manager’s [CRD(Custom Resource Definition)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) ClusterSPIFFEID. It is a cluster-wide resource used to register workloads with SPIRE. The ClusterSPIFFEID can target all workloads in the cluster or can be optionally scoped to specific pods or namespaces via label selectors.
+
+  Create a ClusterSPIFFEID CRD to generate registration entries in SPIRE server for all workloads with the label **`spiffe.io/spire-managed-identity: true`**.
+
+  Get the ClusterSPIFFEID used by us for this demo using **[this link](https://raw.githubusercontent.com/cxteamtrials/caas-trials-content/main/services/spire/clusterspiffeid-example.yaml)**, copy that into your cluster, and apply it.
+
+  ```shellsession
+  kubectl apply -f cluster-spiffeID-crd.yaml
+  ```
+
+  **Note:** You can create your own custom clusterSPIFFEID CRD with your own match label and own selector. For now, we have created simple CRD with one pod selector and one match label.
+
+  Now simply patch the ingress-gateway with spiffe.io/spire managed-identity: true label.
+
+   This will register your ingress-gateway pod into the server.  
+
+  ```shellsession
+  kubectl patch deployment istio-ingressgateway -n istio-system -p '{"spec":{"template":{"metadata":{"labels":{"spiffe.io/spire-managed-identity": "true"}}}}}'
+  ```
+
+  After patching, confirm the working of your ingress-gateway pod, istiod, and all their containers.
+
+## Step 4: Deploying Sample Application
+
+Now that our SPIRE and Istio are integrated, the identities to workloads must be issued by SPIRE. 
+
+For our case, we will create a namespace “bookinfo” and will add a label **“spiffe.io/spire-managed-identity: true”** to it, then we will create a new ClusterSPIFFEID CRD with **namespace selector** with match label as “spiffe.io/spire-managed-identity: true.” 
+
+Now when the new workload is added to this namespace or any other namespace which has the label mentioned above, then automatically it will get registered in the server. 
```

Comment:
Recommend taking out "Now" at the very beginning and rephrasing this sentence to:
When the new workload is added to this namespace or any other namespace that has the lable mentioned above, it will now automatically get registered in the server."

[note the changing of "which" to "that"